ConfigSpace/hyperparameters.pyx

```diff
@@ -1659,6 +2455,44 @@ cdef class CategoricalHyperparameter(Hyperparameter):
                          "OrdinalHyperparameter, but is "
                          "<cdef class 'ConfigSpace.hyperparameters.CategoricalHyperparameter'>")
 
+    def pdf(self, vector: np.ndarray) -> np.ndarray:
+        """
+        Computes the probability density function of the parameter in 
+        the original parameter space (the one specified by the user).
+        For each parameter type, there is also a method _pdf which 
+        operates on the transformed (and possibly normalized) parameter
+        space. Only legal values return a positive probability density,
+        otherwise zero.
+        vector: np.ndarray
+            the (N, ) vector of imputs for which the probability density
+            function is to be computed.
+        """
+        # this check is to ensure shape is right (and np.shape does not work in cython)
+        if vector.ndim != 1:
+            raise ValueError("Method pdf expects a one-dimensional numpy array")
+        vector = self._inverse_transform(vector)
+        return self._pdf(vector)
+        
+    def _pdf(self, vector: np.ndarray) -> np.ndarray:
+        """
+        Computes the probability density function of the parameter in 
+        the transformed (and possibly normalized, depends on the parameter 
+        type) space. As such, one never has to worry about log-normal 
+        distributions, only normal distributions (as the inverse_transform
+         in the pdf method handles these).
+        vector: np.ndarray
+            the (N, ) vector of imputs for which the probability density
+            function is to be computed.
+        """
+        probs = np.array(self.probabilities)
+        res = np.array(probs[vector])
+        if res.ndim == 0:
+            return res.reshape(-1)
```

Comment:
How do you get a 0 dimensional numpy array and what does it mean to reshape it? Don't think I've ever encountered one. If this is the case, can you just return an explicit object, i.e. `np.array([])` or whatever it's meant to be

Reply:
If you feed a 1-dimensional vector of one value here, so e.g.
`vector = np.array([1])` 
which has shape (1, ), and you then retrieve the index, `probs[vector]`, you will actually get the correct value, but in a zero-dimensional vector (so equivalent to `np.array(5)` without brackets). This is a zero-dimensional vector. However, this does not happen if the vector is of any other shape than exactly (1, ).

If you reshape it, you get the desired (1, ) shape (so the same shape as the input for this specific case).

Reply:
Ahh okay, didn't realise you could do `np.array(5)` and that it gives the output it does. Thanks for the pointer, all's good :)